src/wrapper/Plumed.h

```diff
@@ -2874,28 +2955,258 @@ Plumed(Plumed&&p)__PLUMED_WRAPPER_CXX_NOEXCEPT :
+  }
+
+/// cmd_helper with size/data val (typically, std::vector, std::string, small_vector, etc)
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && wrapper::has_size_and_data<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+    std::size_t shape[] { wrapper::size(val), 0 };
+    cmd_helper_with_shape(key,val.data(),shape,nocopy);
+  }
+
+/// cmd_helper with raw pointer val
+/// In this case, the nocopy information is not propagated. We can indeed save the pointer, even if it's a temporary
+/// as it is in the case cmd("a",&a)
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && !wrapper::has_size_and_data<T>::value && std::is_pointer<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+#if __PLUMED_WRAPPER_CXX_DETECT_SHAPES_STRICT
+    // this would be strict checking
+    // "a pointer without a specified size is meant to be pointing to a size 1 object"
+    std::size_t shape[] {  0, 0 };
+    if(val) shape[0]=1;
+    cmd_helper_with_shape(key,val,shape);
+#else
+    // for backward compatibility, the pointer is directly managed by SafePtr, with no size information
+    SafePtr s(val,0,nullptr);
+    cmd_priv(main,key,&s);
+#endif
+  }
+
+/// cmd_helper in remaining cases, that is when val is passed by value
+/// in this case, the nocopy information is not propagated, but the address of the data will be considered not copyable anyway
+/// by PLUMED ("pass by value")
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value && !wrapper::has_size_and_data<T>::value && !std::is_pointer<T>::value, int>::type = 0>
+  void cmd_helper(const char*key,T&& val,bool nocopy=false) {
+    SafePtr s(val,0,nullptr);
+    cmd_priv(main,key,&s);
+  }
+
+/// cmd_helper_with_shape with custom array val (includes std::array and C arrays)
+  template<typename T, typename std::enable_if<wrapper::is_custom_array<T>::value, int>::type = 0>
+  void cmd_helper_with_shape(const char*key,T* val, __PLUMED_WRAPPER_STD size_t* shape,bool nocopy=false) {
+    using value_type = typename wrapper::is_custom_array<T>::value_type;
+    constexpr std::size_t value_size=sizeof(value_type);
+    static_assert(value_size>0,"cannot use custom arrays of void types");
+    static_assert(sizeof(T)%value_size==0,"custom array has incorrect size");
+    auto newptr=reinterpret_cast<value_type*>(val);
+    auto newshape=append_size(shape,sizeof(T)/sizeof(value_type));
+    cmd_helper_with_shape(key,newptr,newshape.data(),nocopy);
+  }
+
+/// cmd_helper_with_shape with pointer to simple type val.
+  template<typename T, typename std::enable_if<!wrapper::is_custom_array<T>::value, int>::type = 0>
+  void cmd_helper_with_shape(const char*key,T* val, __PLUMED_WRAPPER_STD size_t* shape,bool nocopy=false) {
+    SafePtr s(val,0,shape);
+    if(nocopy) s.safe.flags |= 0x10000000;
```

Comment:
the flags modifications may be better enumed or constexpred in order to have names
something like
```suggestion
    if(nocopy) s.safe.flags |= PLMD::SafePtr::SafeFlags::noCopyPolicy;
```
So you can change the flag value in all the possible places where its used as needed, an nearly zero costs

But I think this may be a more "upstream problem"

Reply:
I agree it would be clearer. These are used in C as well, so I cannot use an enum class. Probably macros.

Reply:
And regarding this:
>So you can change the flag value in all the possible places where its used as needed, an nearly zero costs

It should not be changed, because this would break compatibility with previously compiled code. Instead, it is kind of frozen "forever". So, perhaps keeping a number (with a comment) is better, because it increases the cost of changing it :-)